dusty/commands/upgrade.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+import shutil
+from subprocess import check_output, check_call, CalledProcessError
+import tempfile
+import urllib
+
+import psutil
+
+from .. import constants
+from ..daemon import close_client_connection
+from ..log import log_to_client
+from ..payload import daemon_command
+
+
+def _get_latest_version():
+    """Gets latest Dusty binary version using the GitHub api"""
+    url = 'https://api.github.com/repos/{}/releases/latest'.format(constants.DUSTY_GITHUB_PATH)
+    conn = urllib.urlopen(url)
+    if conn.getcode() >= 300:
+        raise RuntimeError('GitHub api returned code {}; can\'t determine latest version.  Aborting'.format(conn.getcode()))
+    json_data = conn.read()
+    return json.loads(json_data)['tag_name']
+
+def _get_binary_url(version):
+    return 'https://github.com/{}/releases/download/{}/{}'.format(constants.DUSTY_GITHUB_PATH, version, constants.DUSTY_BINARY_NAME)
+
+def _get_binary_location():
+    return psutil.Process().exe()
+
+def _download_binary(version):
+    binary_url = _get_binary_url(version)
+    conn = urllib.urlopen(binary_url)
+    if conn.getcode() >= 300:
+        raise RuntimeError('Unable to retrieve Dusty binary version {} from GitHub; this version may not exist'.format(version))
+    binary_data = conn.read()
+    tmp_path = tempfile.mktemp()
+    with open(tmp_path, 'w') as f:
+        f.write(binary_data)
+    os.chmod(tmp_path, 0755)
+    return tmp_path
+
+def _test_dusty_binary(binary_path, version):
```

Comment:
The entire logic is output.split()[-1] assuming the version string is formatted in a particular way.

You're right though, this would make more sense as an integration test.